libs/utils/utils.js

```diff
@@ -621,22 +621,31 @@ function decorateFooterPromo(config) {
   document.querySelector('main > div:last-of-type').insertAdjacentElement('afterend', section);
 }
 
-export function loadIms(onReadyFn) {
-  if (window.adobeIMS) return;
-
-  const { locale, imsClientId, imsScope, env, onReady } = getConfig();
-  if (!imsClientId) return null;
-
-  window.adobeid = {
-    client_id: imsClientId,
-    scope: imsScope || 'AdobeID,openid,gnav',
-    locale: locale?.ietf?.replace('-', '_') || 'en_US',
-    autoValidateToken: true,
-    environment: env.ims,
-    useLocalStorage: false,
-    onReady: onReadyFn || onReady,
-  };
-  loadScript('https://auth.services.adobe.com/imslib/imslib.min.js');
+let imsLoaded;
+export async function loadIms() {
+  imsLoaded = imsLoaded || new Promise((resolve, reject) => {
+    const { locale, imsClientId, imsScope, env } = getConfig();
+    if (!imsClientId) {
+      reject(new Error('Missing IMS Client ID'));
+      return;
+    }
+    const timeout = setTimeout(() => reject(new Error('IMS timeout')), 3000);
```

Comment:
I would say that the IMS timeout should be standardised across consumers, not a configurable value. This will impact Gnav and personalisation, which should be consistent across pages. True to the Milo spirit, I'd say we keep this 'static' until a need for a different behaviour arises.

I'd also argue that a constant might not be needed, since we use the value just once, so we'd end up adding extra characters to `utils` for no real benefit.

One change I would suggest, though, is to raise the value. With a slow 3G connection, Chrome downloads `imslib.js` in about 3.5s (granted, Chrome automatically adds 2s delay in this case if I'm not mistaken), so maybe `4000` would be a better option here.

Reply:
I'll keep as is for now and raised the timeout to 5s. We should observe how this behaves in the real world, also with personalization.
For the GNAV we will have some lana logs coming through seeing when/if/how often this will fail to load on time